test/libsolidity/smtCheckerTests/types/mapping_7.sol

```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: GPL-3.0
+
+
+// Regression for handling signedness, see issue #14792
+contract C {
+    mapping(int => int) v1;
+    mapping(int => uint) v2;
+    mapping(uint => int) v3;
+    mapping(uint => uint) v4;
```

Comment:
What about mixing other types like `mapping(bytes14 => int)` and `mapping(uint => bytes14)`?
Also, I was looking at #14840 and noticed that in that in https://github.com/ethereum/solidity/blob/ae9bcabd954c20e9610a3f3fe64e24cbcf153375/test/libsolidity/smtCheckerTests/types/mapping_6.sol#L6 `v1` is declared and not used. Although this is already merged I was wondering about that.

Reply:
I wanted to suggest a more meaningful name to this test too.
Maybe `mapping_integer_signedness_compatibility/deletion` ... I just think we should not continue the trend of just numbering these tests which contain mappings :)

Reply:
Still regarding the tests, could we also add tests involving arrays, i.e, `int[] x`, `uint[10] y` or does that makes no sense here?

Reply:
> Also, I was looking at #14840 and noticed that in that in
> https://github.com/ethereum/solidity/blob/ae9bcabd954c20e9610a3f3fe64e24cbcf153375/test/libsolidity/smtCheckerTests/types/mapping_6.sol#L6
> 
> `v1` is declared and not used. Although this is already merged I was wondering about that.

The example in #14791 was crashing with this kind of mapping just being declared, but not used. So I kept it in the test.

Reply:
I will rename the test, I agree that it should named properly.

> What about mixing other types like mapping(bytes14 => int) and mapping(uint => bytes14)?

I can add more of these mappings, did not think of `bytes`. The problem was in the type of the domain vs the type of the index. I believe the type of the range should not actually matter.

> Still regarding the tests, could we also add tests involving arrays, i.e, int[] x, uint[10] y or does that makes no sense here?

You mean declaring such arrays and then trying to access/delete some element?

Reply:
> I can add more of these mappings, did not think of `bytes`. The problem was in the type of the domain vs the type of the index. I believe the type of the range should not actually matter.

Ok.

> You mean declaring such arrays and then trying to access/delete some element?

Yes, but I am not sure if that makes sense in the context of the problem here.